homeassistant/components/barry/sensor.py

```diff
@@ -0,0 +1,63 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(hours=1)
+MIN_TIME_BETWEEN_UPDATES = timedelta(hours=1)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the Barry sensor."""
+    barry_connection = hass.data.get(DOMAIN)
+    price_code = hass.data.get("price_code")
+    async_add_entities([BarrySensor(barry_connection, price_code)], True)
+
+
+class BarrySensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, barry_home, price_code):
+        """Initialize the sensor."""
+        self._barry_home = barry_home
+        self.price_code = price_code
+        self._state = None
```

Comment:
```suggestion
```

Reply:
It is used to update the price every hour and change the value of self._state.

Reply:
There is already self._attr_state built-in for that purpose, you don't have to invent new ones.

Reply:
Thanks. I replaced all property methods to class attributes as mentioned, but HA is not able to identify them and displayed `UNKNOWN` state and blank in `ATTRIBUTES` column. I made changes as follows:
Code:
`_attr_name = 'Electricity price Barry'`
`_attr_unit_of_measurement = "DKK/kWh"`
`_attr_state = None`

HA displayes the result as follows:
| Entity  | State | Attibutes |
| ------------- | ------------- | ------------- |
|  sensor.unnamed_device | unknown  | |

Can you please tell me what went wrong here?